runners/spark/src/main/java/org/apache/beam/runners/spark/aggregators/metrics/WithNamedAggregatorsSupport.java

```diff
@@ -0,0 +1,164 @@
+    this.internalMetricRegistry = internalMetricRegistry;
+  }
+
+  public static WithNamedAggregatorsSupport forRegistry(final MetricRegistry metricRegistry) {
+    return new WithNamedAggregatorsSupport(metricRegistry);
+  }
+
+  @Override
+  public SortedMap<String, Timer> getTimers(final MetricFilter filter) {
+    return internalMetricRegistry.getTimers(filter);
+  }
+
+  @Override
+  public SortedMap<String, Meter> getMeters(final MetricFilter filter) {
+    return internalMetricRegistry.getMeters(filter);
+  }
+
+  @Override
+  public SortedMap<String, Histogram> getHistograms(final MetricFilter filter) {
+    return internalMetricRegistry.getHistograms(filter);
+  }
+
+  @Override
+  public SortedMap<String, Counter> getCounters(final MetricFilter filter) {
+    return internalMetricRegistry.getCounters(filter);
+  }
+
+  @Override
+  public SortedMap<String, Gauge> getGauges(final MetricFilter filter) {
+    return
+        new ImmutableSortedMap.Builder<String, Gauge>(
+            Ordering.from(String.CASE_INSENSITIVE_ORDER))
+            .putAll(internalMetricRegistry.getGauges(filter))
+            .putAll(extractGauges(internalMetricRegistry, filter))
+            .build();
+  }
+
+  private Map<String, Gauge> extractGauges(final MetricRegistry metricRegistry,
+                                           final MetricFilter filter) {
+
+    // find the AggregatorMetric metrics from within all currently registered metrics
+    final Optional<Map<String, Gauge>> gauges =
+        FluentIterable
+            .from(metricRegistry.getMetrics().entrySet())
+            .firstMatch(isAggregatorMetric())
+            .transform(toGauges());
+
+    return
+        gauges.isPresent()
+            ? Maps.filterEntries(gauges.get(), matches(filter))
+            : ImmutableMap.<String, Gauge>of();
+  }
+
+  private Function<Map.Entry<String, Metric>, Map<String, Gauge>> toGauges() {
+    return new Function<Map.Entry<String, Metric>, Map<String, Gauge>>() {
+      @Override
+      public Map<String, Gauge> apply(final Map.Entry<String, Metric> entry) {
+        final NamedAggregators agg = ((AggregatorMetric) entry.getValue()).getNamedAggregators();
+        final Map<String, Gauge> gaugeMap = Maps.transformValues(agg.renderAll(), toGauge());
+        return Maps.filterValues(gaugeMap, Predicates.notNull());
+      }
+    };
+  }
+
+  private Function<Object, Gauge> toGauge() {
+    return new Function<Object, Gauge>() {
+
+      @Override
+      public Gauge apply(final Object rawGaugeObj) {
+        return new Gauge<Double>() {
+
+          @Override
+          public Double getValue() {
+            // at the moment the metric's type is assumed to be
+            // compatible with Double. While far from perfect, it seems reasonable at
+            // this point in time
+            try {
+              return Double.parseDouble(rawGaugeObj.toString());
+            } catch (final Exception e) {
+              // log?
```

Comment:
I would WARN about skipping the report of this Aggregator so it won't go undetected. Maybe even with a nice message explaining why.